spring-cloud-kubernetes-client-discovery/src/main/java/org/springframework/cloud/kubernetes/client/discovery/KubernetesClientInformerAutoConfiguration.java

```diff
@@ -57,6 +58,7 @@
 @ConditionalOnKubernetesDiscoveryEnabled
 @ConditionalOnBlockingOrReactiveEnabled
 @ConditionalOnCloudPlatform(CloudPlatform.KUBERNETES)
```

Comment:
this one is only active when `spring.cloud.kubernetes.discovery.namespaces` is not present.